README.md

```diff
@@ -108,43 +108,86 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Hazelcast serializes all your objects into an instance of `Data`. `Data` is the binary representation of an object. When Hazelcast serializes an object into `Data`, it first checks whether the object is an instance of `IdentifiedDataSerializable`, if not it checks if it is an instance of `Portable` and serializes it accordingly. For the following types Hazelcast optimizes the serialization a user can not override this behaviour.
+- A `boolean` type which is a primitive on Node.js can be serialized as a `Boolean` for Java server-side.
```

Comment:
I am guessing there is a new line before the first bulleted item.

...a primitive on the Node.js client can be serialized as a `Boolean` for the Java server side.

(need to apply to below "server-side" since it is not used as an adjective here)